src/engine/Server.cpp

```diff
@@ -1037,12 +1031,5 @@ std::optional<std::string> Server::checkParameter(
   } else if (value != parameterValue) {
     return std::nullopt;
   }
-  // Now that we have the value, check if there is a problem with the access.
-  // If yes, we abort the query processing at this point.
-  if (!accessAllowed) {
-    throw std::runtime_error(absl::StrCat(
-        "Access to \"", key, "=", value.value(), "\" denied",
-        " (requires a valid access token)", ", processing of request aborted"));
-  }
```

Comment:
I don't see the rationale behind moving this into a convoluted lambda, 
can you please explain or revert?

Reply:
There was a lambda in processQueryOrUpdate that did something very similar to `Server::checkParameter`. The access logic was only required here. So I moved that out of the function into the lambda (which existed before).

Proposal for moving forward here: the accessAllowed is used for certain functionality which requires a valid access token. Having this in a function that extracts parameters or checks if parameters has some value is weird. The access checking becomes a new function/lambda `requireValidAccessToken`. The `checkParameter` lambda would then disappear.